### ISSUE TITLE
Naming convention

### ISSUE THREAD

**Issue by verri** [verri/dnaya#1]
While graph transformation techniques have consistent naming, most of the dynamical processes used in machine learning do not.

For instance, [Verri, Gueleri, et al. (2018)](https://arxiv.org/abs/1802.04186) do not name their proposed flocking-based system. Calling it "flocking" is too general. "Particle alignment" also seems too vague.

For those systems, I propose (maybe temporarily?) naming them with the initials of the authors. (Systems that already have a name, we keep them.) Thus, the system in

- [Verri, Gueleri, et al. (2018)](https://arxiv.org/abs/1802.04186) becomes `vgz3`
- [Pellicer, Verri, and Curtis (2019)](https://doi.org/10.1109/BRACIS.2019.00142) is `lcu` (Labeled Component Unfolding)
- [Verri and Zhao (2016)](https://doi.org/10.1109/bracis.2016.051) is `rw` (Random Walk)

The mutualistic system is still unpublished, so we can decide that later.  Suggestions are `ms` (maybe too general) or `vjzl` (Verri, Jiang, Zhao, and Lai).

**Comment by ryukinix** [comment]
I like auto-explainable names for things, like convolution instead of green's theorem. As always as possible, we should use that easier names to avoid confusion later. Naming things is hard, but refactoring name after having users it's even more harder. 

Random Walk and Labeled Component Unfolding are good names. About `vgz3` and `vzjl` we can try find a better one for them, because they are very unfriendly for new users.

About mutualistic system, `ms` it's a very common term that can be confused easily. The mutualistic system name itself it's generic as you said.

I need to think about that naming stuff in general. I'll read each one of the articles where is proposed that systems and I'll try suggest a name for us.

**Comment by verri** [comment]
Well, you're right, naming is hard indeed.

Actually, even the random walk in Verri and Zhao (2016) is not a traditional random walk.

I think we can brainstorm names for each system besides LCU (the only one that has a name, believe it or not.)

Some info about the systems:

- The system in Verri, Gueleri, et al. (2018) is literally a process of _alignment of "coupled" multidimensional vectors_.
Although, I am not sure "coupled" works in this context. (I borrowed the term from oscillators' systems.)
- The system in Verri and Zhao (2016) is a random walk in a bipartite network with some preferential edges.
- The system in [Uzun and Ribeiro (2017)](https://doi.org/10.1371/journal.pone.0182737) is a naming game with three social features: trust, uncertainty, and opinion preference.
- The mutualistic system is a population evolution of species with mutualistic relationships.

We can also take into consideration the *effect* (or the purpose) of the system to decide a name, like LCU.

**Comment by ryukinix** [comment]
All right! Brainstorming for that case is a good idea.

> We can also take into consideration the effect (or the purpose) of the system to decide a name, like LCU.

Yes, I agree on that.

Let's try it. Some names I thought:

- Verri, Gueleri, et al. (2018):
  - Coupled Vectors Alignment (CVA)
- Verri and Zhao (2016):
  - Selective Random Walk (SRW)
  - Bipartite Random Walk (BRW)
- Uzun and Ribeiro (2017):
  - Opinionated Community Detector (OCD)
  - Social Affinity Community Detector (SACD)
- Mutualistic System:
  - Mutualistic Evolution (ME)
  - Mutualistic Generation (MG)
  - Mutualistic Optimization (MO)

**Comment by verri** [comment]
> Coupled Vectors Alignment (CVA)

+1

> Selective Random Walk (SRW)
> Bipartite Random Walk (BRW)

Too general, conflicts with other names in literature.
I suggest "Non-uniform Random Walk (NRW)".
(The dynamics can be slightly changed to work on other types of networks.)

> Opinionated Community Detector (OCD)
> Social Affinity Community Detector (SACD)

Social Affinity Partitioning (SAP)
(Community detection seems too specific.)

About the mutualistic system, I suggest Mutualistic Species Survival (MSS).
Later, I can explain to you the details of this system and the reasoning for its name.

**Comment by ryukinix** [comment]
> > Coupled Vectors Alignment (CVA)
> 
> +1

Nice! CVA choosed.

> 
> > Selective Random Walk (SRW)
> > Bipartite Random Walk (BRW)
> 
> Too general, conflicts with other names in literature.
> I suggest "Non-uniform Random Walk (NRW)".
> (The dynamics can be slightly changed to work on other types of networks.)
> 

I agree, it's general, but Non-uniform maybe is not the best word yet, let's try using https://www.thesaurus.com/browse/nonuniform


> > Opinionated Community Detector (OCD)
> > Social Affinity Community Detector (SACD)
> 
> Social Affinity Partitioning (SAP)
> (Community detection seems too specific.)

Social Affinity Partitioning it's ok to me, the good part is the literature usage on graph theory, but partitioning has a bad spelling (easy to mispelling). Some other alternatives for partitioning: https://www.thesaurus.com/browse/partition?s=t

- Social Affinity Segregation
- Social Affinity Clustering
- Social Affinity Splitting

> 
> About the mutualistic system, I suggest Mutualistic Species Survival (MSS).
> Later, I can explain to you the details of this system and the reasoning for its name.

Ok! I'd like to hear!

**Comment by verri** [comment]
> I agree, it's general, but Non-uniform maybe is not the best word yet, let's try using https://www.thesaurus.com/browse/nonuniform

None of the alternatives have the same meaning. We can also consider a diffusion/attraction process instead of a preferential random walk. In practice, they are the same system. One problem that I see in such an approach is that we should avoid the term "attractor" since it has another meaning in Complex Systems. Let's mature the idea...

> Social Affinity Segregation

I like this one, but the acronym, SAS, is not a good choice.

>  Social Affinity Clustering

Clustering seems too specific.

**Comment by verri** [comment]
What about Biased Random Walk (BRW)? I like the acronym.

**Comment by verri** [comment]
> Social Affinity Clustering

- Social Affinity Formation
- Social Affinity Arrangement
- Social Affinity Mapping
- Social Affinity Grouping

**Comment by ryukinix** [comment]
> > I agree, it's general, but Non-uniform maybe is not the best word yet, let's try using https://www.thesaurus.com/browse/nonuniform
> 
> None of the alternatives have the same meaning. We can also consider a diffusion/attraction process instead of a preferential random walk. In practice, they are the same system. One problem that I see in such an approach is that we should avoid the term "attractor" since it has another meaning in Complex Systems. Let's mature the idea...

Ok!

> 
> > Social Affinity Segregation
> 
> I like this one, but the acronym, SAS, is not a good choice.

Yes, SAS is not good.
> 
> > Social Affinity Clustering
> 
> Clustering seems too specific.

Ok

> What about Biased Random Walk (BRW)? I like the acronym.

I thought about that earlier, but Biased doesn't sounds bad? In various context of statistics and machine learning bias it is a problem. Maybe Guided Random Walk? Semi Blind Walk? I don't know. Let's try a little more.

> > Social Affinity Clustering
> 
> * Social Affinity Formation
> * Social Affinity Arrangement
> * Social Affinity Mapping
> * Social Affinity Grouping

I really liked Social Affinity Grouping.

**Comment by ryukinix** [comment]
What we have until now (feel free to edit my comment cell):

| article | algorithm name | acronym|
|----------|----------------------|--------------|
| Verri and Zhao (2016) | Biased Random Walk | BRW|
| Uzun and Ribeiro (2017) | Social Affinity Grouping | SAG|
| Verri, Gueleri, et al. (2018) | Coupled Vectors Alignment | CVA|
| Mutualistic article (ref?) | Mutualistic Species Survival | MSS|

That names are looking really good! Brainstorm has been productive!

**Comment by verri** [comment]
> I thought about that earlier, but Biased doesn't sounds bad? In various context of statistics and machine learning bias it is a problem. Maybe Guided Random Walk? Semi Blind Walk? I don't know. Let's try a little more.

Indeed, statistical bias sounds negative. However, in machine learning, it is a misconception to see bias as a negative thing. "No [bias](https://en.wikipedia.org/wiki/Inductive_bias), no learning". Also, these methods are transductive and nonparametric. In this context, the meaning of the term "bias" bears no relation whatsoever to the [bias of an estimator"](https://en.wikipedia.org/wiki/Bias_of_an_estimator).

**Comment by ryukinix** [comment]
> > I thought about that earlier, but Biased doesn't sounds bad? In various context of statistics and machine learning bias it is a problem. Maybe Guided Random Walk? Semi Blind Walk? I don't know. Let's try a little more.
> 
> Indeed, statistical bias sounds negative. However, in machine learning, it is a misconception to see bias as a negative thing. "No [bias](https://en.wikipedia.org/wiki/Inductive_bias), no learning". Also, these methods are transductive and nonparametric. In this context, the meaning of the term "bias" bears no relation whatsoever to the [bias of an estimator"](https://en.wikipedia.org/wiki/Bias_of_an_estimator).

That's true! With that in my mind, now I think biased may work well.
